specs/SIM/alert_manager/SODA_AlertManagerDesign.md

```diff
@@ -0,0 +1,239 @@
+| Input     | Details of alarm to be cleared  |
+| Key Processings     | Event received by alert config |
+|                     | Parse and get details of alarm to be cleared |
+|                     | Trigger alarm clear from back end |
+| Output    | Input alarm is cleared at backend  |
+| Priority     | High |
+| Remarks     | None |
+
+##### AlertMngr.Int.004 Add alert source configuration
+|  |  |
+|-----------------|---------------------------|
+| Brief Description    | Configuring alert source for SNMP V2c/V3 user for processing traps |
+| Precondition       | Module is up |
+| Input     | Alert source configuration from admin  |
+| Key Processings     | Validate input alert source wrt its version |
+|                     | Add alert source details to DB |
+|                     | Update trap receiver with alert source config |
+| Output    | Input alert source is saved to database and trap receiver is updated accordingly to handle traps |
+| Priority     | High |
+| Remarks     | None|
+
+##### AlertMngr.Int.005 List alert source configuration
+|  |  |
+|-----------------|---------------------------|
+| Brief Description    | Listing all configured alert sources |
+| Precondition       | Module is up |
+| Input     | List alert source request from admin  |
+| Key Processings     | Validate input query request |
+|                     | Return the list of all alert sources from DB |
+| Output    | List of all alert sources|
+| Priority     | High |
+| Remarks     | None|
+
+##### AlertMngr.Int.006 Delete alert source configuration
+|  |  |
+|-----------------|---------------------------|
+| Brief Description    | Deleting already configured alert source |
+| Precondition       | Module is up |
+| Input     | Delete Alert source configuration request from admin  |
+| Key Processings     | Validate input alert source delete request |
+|                     | Delete alert source details from DB |
+|                     | Remove alert source config details from trap receiver side |
+| Output    | Input alert source configuration is deleted from database |
+| Priority     | High |
+| Remarks     | None|
+
+##### AlertMngr.Int.007 Update alert source configuration
+|  |  |
+|-----------------|---------------------------|
+| Brief Description    | Updating existing alert source configuration |
+| Precondition       | Module is up |
+| Input     | Update Alert source configuration requestfrom admin  |
+| Key Processings     | Validate input alert source wrt its version |
+|                     | Update alert source details to DB |
+|                     | Update trap receiver with alert source config |
+| Output    | Input alert source is updated to database and trap receiver is updated accordingly to handle traps |
+| Priority     | High |
+| Remarks     | None|
+
+
+#### Non Functional Requirements
+NA
+
+## Architecture Analysis
+### SODA Infrastructure Manager: High Level View
+![](../resources/Architecture.jpg)
+
```

Comment:
What is our considerations?
- Can the Alert Manager and Driver Manager be separate services running independently? Say alert Manager is running on node1 and device manager on Node2..
- The duty of "Alert config" is just to trap config and clear alert or something more? 
- Can alert config be part of Device Manager as the task currently it is doing is just to be a router